src/components/inline/Button/Button.js

```diff
@@ -3,18 +3,59 @@ import React from "react";
 
 import s from "./Button.module.css";
 
-const Button = ({ children }) => (
-  <button className={s.button} type="button">
-    {children}
-  </button>
-);
+const Button = ({ button }) => {
+  let buttonView;
 
-Button.propTypes = {
-  children: PropTypes.node,
+  if (button.externalLink) {
+    buttonView = (
+      <a rel="noreferrer" target="_blank" href={button.externalLink}>
+        <button type="button" className={s.button}>
+          {button.text}
+        </button>
+      </a>
+    );
+  } else if (button.internalLink) {
+    buttonView = (
+      <a href={button.internalLink}>
+        <button type="button" className={s.button}>
+          {button.text}
+        </button>
+      </a>
+    );
+  } else {
+    buttonView = (
+      <button
+        type="button"
+        className={s.button}
+        onClick={() => {
+          button.scriptAction();
+        }}
+      >
+        {button.text}
+      </button>
+    );
+  }
+
+  return <div>{buttonView}</div>;
 };
 
-Button.defaultProps = {
-  children: null,
+const ButtonPropType = PropTypes.oneOfType([
+  PropTypes.exact({
+    text: PropTypes.string,
+    externalLink: PropTypes.string,
+  }),
+  PropTypes.exact({
+    text: PropTypes.string,
+    internalLink: PropTypes.string,
+  }),
+  PropTypes.exact({
+    text: PropTypes.string,
+    scriptAction: PropTypes.func,
+  }),
+]);
+
+Button.propTypes = {
+  button: ButtonPropType.isRequired,
```

Comment:
This extra layer seems unnecessary. Why not just set `Button.propTypes = PropTypes.oneOfType(...)`? Otherwise, you have this extra layer of thing that you have to step through in order to access values.